File: src/main/java/com/cfgglobal/generator/entity/Task.java
package com.cfgglobal.generator.entity;


import com.cfgglobal.generator.task.processor.ITaskProcessor;
import lombok.Data;
import lombok.extern.slf4j.Slf4j;
import org.apache.commons.lang3.StringUtils;
import org.joor.Reflect;

import java.util.List;
import java.util.Map;

@Data
@Slf4j
public class Task {
    private Integer id;
    private String name;
    private String taskType;
    private String folder;
    private String filename;
    private String templatePath;

    public List<String> run(CodeProject codeProject, Map<String, Object> root) {
        log.debug("task " + this.getName() + " run");
        List<String> paths = taskProcessor(this.getTaskType()).run(codeProject, this, root);
        log.debug("task " + this.getName() + " end");
        return paths;
    }

    private ITaskProcessor taskProcessor(String taskType) {
        return Reflect.on(ITaskProcessor.class.getPackage().getName() + "."
                + StringUtils.capitalize(taskType) +
                "TaskProcessor").create().get();
    }
}

File: src/main/kotlin/com/github/leon/generator/entity/Task.kt
package com.github.leon.generator.entity


import com.github.leon.generator.task.processor.ITaskProcessor
import org.apache.commons.lang3.StringUtils
import org.joor.Reflect


data class Task(
        var id: Int? = null,

        var name: String,

        var taskType: String,

        var taskOfProject:TaskOfProject,

        var folder: String,

        var filename: String,

        var templatePath: String,

        var replaceFile: Boolean = true
) {

    fun run(codeProject: CodeProject, root: MutableMap<String, Any>): List<String> {
        return taskProcessor(taskType).run(codeProject, this, root)
    }

    private fun taskProcessor(taskType: String): ITaskProcessor {
        return Reflect.on(ITaskProcessor::class.java.`package`.name + "."
                + StringUtils.capitalize(taskType) +
                "TaskProcessor").create().get()
    }


}

